galley/pkg/server/server.go

```diff
@@ -172,7 +172,13 @@ func newServer(a *Args, p patchTable) (*Server, error) {
 			Time:                  a.KeepAlive.Time,
 			MaxConnectionAge:      a.KeepAlive.MaxServerConnectionAge,
 			MaxConnectionAgeGrace: a.KeepAlive.MaxServerConnectionAgeGrace,
-		}))
+		}),
+		// Relax keepalive enforcement policy requirements to avoid dropping connections due to too many pings.
+		grpc.KeepaliveEnforcementPolicy(keepalive.EnforcementPolicy{
+			MinTime:             time.Minute,
+			PermitWithoutStream: true,
```

Comment:
// If true, client sends keepalive pings even with no active RPCs. If false,
	// when there are no active RPCs, Time and Timeout will be ignored and no
	// keepalive pings will be sent.
	PermitWithoutStream bool // false by default.